### PR TITLE
Build scripts: Name client variants by their build system for clarity

### PR DESCRIPTION
**Pull request by juliettepretot**
TL;DR

`cpp` -> `bazel`
`rust` -> `cargo`
`nodejs` -> `npm`

----

Addresses @daviddrysdale's earlier comment: https://github.com/project-oak/oak/pull/1169#discussion_r442165207

This renames the issue of the somewhat confusing `cpp` variant. 

This makes sense, since bazel is the build system the scripts will use for this variant. And other non cpp clients (Go) that use bazel shouldn't use the `cpp` flag.

Similarly this renames `rust` to `cargo`, and `nodejs` to `npm`, so we always refer to the build system. 

This is quite similar to the naming used in the rust runner. 
https://github.com/project-oak/oak/blob/19ff6bb2c676f501816d8a57c3f430f5bed45463/runner/src/main.rs#L295

Imho it might be worth to think about doing the same for the server/application variants.

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: scripts/run_example [hunk 1]
```diff
@@ -123,18 +123,18 @@ do
         echo "The ${EXAMPLE} example does not contain a ${client_language} client. Skipping this client."
       fi
       ;;
-    cpp)
-      cpp_client="./bazel-client-bin/examples/${EXAMPLE}/client/client"
+    bazel)
+      cpp_client="${SCRIPTS_DIR}/../examples/${EXAMPLE}/client/BUILD"
```
Comment (juliettepretot) [inline scripts/run_example:127]: This also means we now explicitly check for the presence of a BUILD file.

File: scripts/run_example [hunk 2]
```diff
@@ -27,18 +27,18 @@ Options:
           - rust (used by default)
           - cpp
   -c    Example client variants. Multiple clients can be specified by space
-        seperating them. Eg -c \"rust cpp nodejs\".
-          - rust
-          - cpp (used by default)
-          - nodejs
+        seperating them. Eg -c \"cargo bazel npm\".
```
Comment (daviddrysdale) [inline scripts/run_example:30]: pre-existing nit: s/seperating/separating/

File: scripts/run_examples [hunk 3]
```diff
@@ -18,16 +18,16 @@ while getopts "s:a:c:h" opt; do
           - rust (default)
           - cpp
   -c    Example client variants. Multiple clients can be specified by space
-        seperating them. Eg -c \"rust cpp nodejs\".
-        - rust
-        - cpp (used by default)
-        - nodejs
+        seperating them. Eg -c \"cargo bazel npm\".
```
Comment (daviddrysdale) [inline scripts/run_examples:21]: pre-existing nit: s/seperating/separating/
